src/index.js

```diff
@@ -50,18 +56,8 @@ export function genericHashLink(props, As) {
   function handleClick(e) {
     reset();
     if (props.onClick) props.onClick(e);
-    if (typeof props.to === 'string') {
-      hashFragment = props.to
-        .split('#')
-        .slice(1)
-        .join('#');
-    } else if (
-      typeof props.to === 'object' &&
-      typeof props.to.hash === 'string'
-    ) {
-      hashFragment = props.to.hash.replace('#', '');
-    }
-    if (hashFragment !== '') {
+    hashFragment = props.to.hash || props.to;
```

Comment:
Thanks for the PR. I like this and will add it.
However.. this needs better type checking here to make sure hashFragment is a string. For example:
```js
// if
props.to = {
  pathname: '/some-path',
}

// then
hashFragment = props.to;
// which is an object and causes problems in getElAndScroll
// where string methods are run on it
```

I'd probably just keep the deleted `if/else if` type checking logic (only remove the string manipulation).